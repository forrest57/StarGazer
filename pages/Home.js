import { StatusBar } from 'expo-status-bar'
import { useState } from 'react'
import { StyleSheet, Text, View, SafeAreaView, TextInput } from 'react-native'
import tw from 'twrnc'
import AnimatedButton from '../components/animatedButton'
import GazerCard from '../components/gazerCard'

export default Home = () => {
  // const data = [
  //   { id: 1, text: 'loremipsum' },
  //   { id: 5, text: 'loremipsum' },
  //   { id: 3, text: 'loremipsum' },
  //   { id: 2, text: 'loremipsum' },
  //   { id: 4, text: 'loremipsum' },
  //   { id: 6, text: 'loremipsum' },
  //   { id: 7, text: 'loremipsum' },
  //   { id: 8, text: 'loremipsum' },
  //   { id: 9, text: 'loremipsum' },
  //   { id: 10, text: 'loremipsum' },
  // ]
  const [repo, setRepo] = useState('')
  const [user, setUser] = useState('')
  //required in order to not have to destructure props multiple times
  const cardRenderer = ({ item }) => <GazerCard text={item.text} />

  return (
    <SafeAreaView
      style={[
        { marginTop: StatusBar.currentHeight || 0 },
        tw`flex flex-col items-center justify-center bg-gray-800 h-full  `,
      ]}>
      <Text style={styles.title}>Stargazer</Text>
      <View style={tw`flex flex-row w-3/4 items-center justify-between`}>
        <TextInput
          autoFocus={true}
          cursorColor='white'
          style={[tw` w-5/12 p-3 bg-gray-700  rounded-md`, styles.TextInput]}
          placeholder='User'
          placeholderTextColor={'gray'}
          onChangeText={(newUser) => setUser(newUser)}
        />
        <Text style={[{ fontSize: 46, color: 'gray' }, tw`w-1/12 `]}>/</Text>
        <TextInput
          cursorColor='white'
          style={[tw` w-5/12 p-3 bg-gray-700  rounded-md`, styles.TextInput]}
          placeholder='Repo'
          onChangeText={(newRepo) => setRepo(newRepo)}
          placeholderTextColor={'gray'}
        />
      </View>
      <AnimatedButton text={'Look gazers up'} />
      <AnimatedButton text={'Recent repos'} />
      {/* <Animated.View style={[{ transform: [{ scale }] }, tw`w-3/4 my-2 flex `]}>
        <Pressable
          activeOpacity={1}
          onPressIn={onPressIn}
          onPressOut={onPressOut}
          onPress={onPress}
          style={tw`bg-gray-700 p-2 w-full rounded-md flex items-center`}>
          <Text style={styles.buttonText}>let's gooo</Text>
        </Pressable>
      </Animated.View> */}
      {/* <View style={tw`flex flex-row bg-gray-700 p-1  m-2 mt-7 rounded-md`}>
        <TextInput
          autoFocus={true}
          cursorColor='white'
          style={[tw`  w-6/7 p-3`, { fontSize: 28, color: 'white' }]}
          placeholder='User/repo'
        />
        <Pressable style={tw` flex items-center justify-center flex-1`}>
          <FontAwesome name='search' size={32} color='gray' />
        </Pressable>
      </View>
      <FlatList
        data={data}
        renderItem={cardRenderer}
        keyExtractor={(item) => item.id}
      /> */}
      <StatusBar style='auto' />
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({
  TextInput: {
    fontSize: 28,
    color: 'white',
  },
  title: {
    fontSize: 46,
    color: 'gray',
    fontWeight: 'bold',
    fontStyle: 'italic',
  },
  buttonText: {
    fontSize: 36,
    color: 'gray',
    fontWeight: 'bold',
  },
})
