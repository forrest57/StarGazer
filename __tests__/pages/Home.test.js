import React from 'react'
import { TextInput } from 'react-native'
import renderer from 'react-test-renderer'
// import App from '../App'
import Home from '../../pages/Home'

describe('<Home />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  // it('correctly updates state', () => {
  //   const setRepo = jest.fn()
  //   const tree = shallow(<Home />)
  //   console.log(tree)
  //   tree.findByType(<TextInput />).simulate('changeText', 'loremipsum')
  //   expect(setRepo).toHaveBeenCalledWith('loremipsum')
  // })
})
