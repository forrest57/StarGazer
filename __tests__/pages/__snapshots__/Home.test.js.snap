// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "marginTop": 0,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#1f2937",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 46,
        "fontStyle": "italic",
        "fontWeight": "bold",
      }
    }
  >
    Stargazer
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "75%",
      }
    }
  >
    <TextInput
      autoFocus={true}
      cursorColor="white"
      onChangeText={[Function]}
      placeholder="User"
      placeholderTextColor="gray"
      style={
        Array [
          Object {
            "backgroundColor": "#374151",
            "borderRadius": 6,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": "41.666667%",
          },
          Object {
            "color": "white",
            "fontSize": 28,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "gray",
            "fontSize": 46,
          },
          Object {
            "width": "8.333333%",
          },
        ]
      }
    >
      /
    </Text>
    <TextInput
      cursorColor="white"
      onChangeText={[Function]}
      placeholder="Repo"
      placeholderTextColor="gray"
      style={
        Array [
          Object {
            "backgroundColor": "#374151",
            "borderRadius": 6,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": "41.666667%",
          },
          Object {
            "color": "white",
            "fontSize": 28,
          },
        ]
      }
    />
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "display": "flex",
        "marginTop": 8,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": "75%",
      }
    }
  >
    <View
      accessible={true}
      activeOpacity={1}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#374151",
          "borderRadius": 6,
          "display": "flex",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 32,
            "fontWeight": "bold",
          }
        }
      >
        Check Gazers
      </Text>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "display": "flex",
        "marginTop": 8,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": "75%",
      }
    }
  >
    <View
      accessible={true}
      activeOpacity={1}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#374151",
          "borderRadius": 6,
          "display": "flex",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 32,
            "fontWeight": "bold",
          }
        }
      >
        Recent repos
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
